usr/sbin/so-elastic-configure-stack

```diff
@@ -37,27 +37,40 @@ for USER in elasticsearch logstash kibana elastalert curator freqserver domainst
 done
 
 header "Configuring ElasticSearch"
-mkdir -p /etc/elasticsearch
-mkdir -p /nsm/elasticsearch
-mkdir -p /var/log/elasticsearch
-chown -R elasticsearch:elasticsearch /nsm/elasticsearch
-chown -R elasticsearch:elasticsearch /var/log/elasticsearch
+
+# Create Elasticsearch directories
+for i in /etc/elasticsearch/ /nsm/elasticsearch /var/log/elasticsearch; do
+	[ ! -d $i ] && mkdir -p $i
+done
+
 touch /etc/nsm/crossclustertab
-chown elasticsearch:elasticsearch /etc/nsm/crossclustertab 
-cp -av $SRC/etc/elasticsearch/* /etc/elasticsearch/
-cp -av $SRC/etc/logrotate.d/crossclustercheck /etc/logrotate.d/crossclustercheck
-if [ -f /etc/elasticsearch/elasticsearch.yml ]; then
-        sed -i "s/cluster\.name.*/cluster\.name: \"$HOSTNAME\"/" /etc/elasticsearch/elasticsearch.yml
+
+for i in /nsm/elasticsearch /var/log/elasticsearch /etc/nsm/crossclustertab; do
+	chown -R elasticsearch:elasticsearch $i
+done
+
+cp -av $SRC/etc/elasticsearch/log4j2.properties /etc/elasticsearch/log4j2.properties
+
+ES_YAML="/etc/elasticsearch/elasticsearch.yml"
+if [ -f "$ES_YAML" ]; then
+	# Preserve transport settings if present
+	if grep -q transport $ES_YAML; then
+		:
+	else
+		cp -av $SRC/$ES_YAML $ES_YAML
```

Comment:
If I'm reading this correctly, then if ES_YAML does NOT exist (new installation), it will never get copied into place.  Is that correct?

Maybe we should do something like this?

if $ES_YAML exists, copy to $ES_YAML_BAK

cp -av $SRC/$ES_YAML $ES_YAML

if $ES_YAML_BAK exists and contains transport settings, then copy those settings to $ES_YAML